### DISCUSSION TITLE
✅ Add support to AVIF for Elementor Lightbox.

### DISCUSSION THREAD

**Discussion by bona-mkl** [elementor/elementor#25175] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

I created a php function to add AVIF image file type to WP. Elementor works okay, and it recognizes it as image file. But, when using Gallery widget to open images in Light Box, it "downloads" the image file instead of opening in the lightbox. The fun thing, is that when I add other type of image file like JPG, PNG or WEBP, the lightbox opens normally, and then I can navigate the entire gallery using arrows and it shows all the AVIF type images too. 

### Describe the solution you'd like.

It would be nice if the lightbox opens the AVIF type images instead of downloading it.

### Describe alternatives you've considered.

I could use WEBP images but it is bigger than AVIF and the image compression result is not good. AVIF is better type.

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by Angelo-Verona** [comment]
Yes, this is vital for some of us, we are using only avif nowadays but just for a gallery we are forced to use webP. Its that harder that we automatically change uploads to avif, and during uploading of images to gallery we need to switch that for whole website to webP. 
Would be good to have some settings where we can map image formats, or something.

**Reply by bona-mkl** [reply]
One solution is to use a third party lightbox plug-in that has AVIF type file support, and deactivate the Elementor built-in lightbox. In my case, I created a [shortcode] and added into a simple popup without the funcions like zoom, only to show the whole image.

**Comment by bona-mkl** [comment]
After visiting Elementor Facebook Group and Discord, I realized that this matter is not a priority to Elementor, since they created the "Image Optimization" plug-in. Therefore, in WP you can find diverse third-party lightbox plugins that has AVIF file type support.

**Reply by m1ga** [reply]
https://github.com/elementor/elementor/blob/main/assets/dev/js/frontend/utils/lightbox/lightbox-manager.js#L31 is the place where you would need to add avif. You can edit your local `/elementor/assets/js/frontend.min.js` and add it there to get it working. Just make sure to edit it when you update Elementor again

**Reply by m1ga** [reply]
PR https://github.com/elementor/elementor/pull/28256

**Reply by Nazrinn** [reply]
This is still an issue. I have AVIF images on my site and it's breaking my lightboxes. @m1ga I has to implement your fix. I see that you have done a pull request about it that seems to have been implemented. Was your change reversed?

**Reply by m1ga** [reply]
it was merged into master, so it's not released yet

**Reply by Nazrinn** [reply]
Does this mean the fix will be out next update?

**Reply by m1ga** [reply]
I only made the PR (over half a year ago), I have nothing to do with Elementor or know their release cycle or if it will make it into the next release, sorry. I only know it was merged into main and tagged with 3.29.0. But if it really will be in it, I don't know. They can change it or not.

**Reply by Nazrinn** [reply]
Roger that.

**Reply by bona-mkl** [reply]
Good to know. Thank you!

**Comment by Nazrinn** [comment]
For anyone looking for a temporary fix, you can do the follwing on your website:
Login to WordPress dashboard > Plugin > Plugin file editor > Access the elementor directory > In assets/js/frontend.min.js, find the line that says `(png|jpe?g|gif|svg|webp)` with Ctrl + F (page search) and modify the line to` (png|jpe?g|gif|svg|webp|avif)` > Save. This will have to be repeated every time you update the Elementor plugin.

![image](https://github.com/user-attachments/assets/a329974f-418b-4d22-8dd5-d39c7f5ecb11)

**Reply by Nazrinn** [reply]
@StyleShit Sorry for the ping, but since I know you're an Elementor employee I wanted to get your attention. This is the second time this bug breaks my client's website and the fix required is really minor and already targeted by a contributor. I really hope this get fixed by next update.

**Reply by m1ga** [reply]
as mentioned before: it is already merged and will be in the next version. You can download the beta here https://github.com/orgs/elementor/discussions/31023 and see it's in there. Current release date is `May 19th` (can change during the beta test).

**Reply by abdulmecidd** [reply]
This solution was fixed my problem.

**Reply by bona-mkl** [reply]
> as mentioned before: it is already merged and will be in the next version. You can download the beta here https://github.com/orgs/elementor/discussions/31023 and see it's in there. Current release date is `May 19th` (can change during the beta test).

It's today! Hope to see it working finally! :)

**Reply by m1ga** [reply]
it is, but it would be very beneficial to test the beta version that was released 3 weeks ago for such issues in order to notify them if it wasn't working and then they could make fixes. In this case my fix was only 5 characters so that hopefully works for you too :smile: If so: you can close this

**Reply by bona-mkl** [reply]
I've just tested this feature at Elementor Ver 3.29.0 and it's working now. 😃 Thank you Elementor team. Also this new version looks so great!

**Comment by bona-mkl** [comment]
I can definitely do it, but within next week. Once I check it's working, I'll close the Discussion for sure. 😊

[cid:60d4d3e0-1ecb-409a-84a1-867883945d38]<https://bonalee.work/?utm_source=bnl_hotmail&utm_medium=email>



________________________________
De: Michael Gangolf ***@***.***>
Enviadas: Segunda-feira, 19 de Mai de 2025 11:25
Para: elementor/elementor ***@***.***>
Cc: Bona, Min K. Lee | 이 민경 ***@***.***>; Author ***@***.***>
Assunto: Re: [elementor/elementor] Add support to AVIF for Elementor Lightbox. (Discussion #25175)


it is, but it would be very beneficial to test the beta version that was released 3 weeks ago for such issues in order to notify them if it wasn't working and then they could make fixes. In this case my fix was only 5 characters so that hopefully works for you too 😄 If so: you can close this

**Comment by bona-mkl** [comment]
This feature now available on Elementor Ver 3.29.0 👍

**Comment by nicholaszein** [comment]
Hey everyone!

We have great news! 🙌 

### 📢 We're happy to announce that this feature was released with Elementorv3.29.0! 🥳 

Elementor's Lightbox now supports **AVIF** image formats natively!

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
